### PR TITLE
Update cortx-monitorQuickstartGuide.md

### PR DESCRIPTION
**Pull request by thavanathan**
## Coding:

* [ ] Did you follow coding standard and verified it with flake8? 

* [ ] Did your address all Codacy issues?

## Testing:

* [ ] Unittest updated and all unittests are passing?

* [ ] Sanity/Self tests are updated (if applicable)?

* [ ] Manual testing done covering happy path and non-happy path?

## Integration:

* [ ] If there is any interface change, did you communicate it to other components?
* [ ] If there is any interface change, other componenent gate gatekeepers are ready to accept the change?
* [ ] Did you complete deployment test if applicable?

## PR checklist:
* [ ] Did you add Jira number to the PR?

* [ ] DCO and cla-signed?

-----
[View rendered cortx-monitorQuickstartGuide.md](https://github.com/Seagate/cortx-monitor/blob/thavanathan-patch-1/cortx-monitorQuickstartGuide.md)

### CODE REVIEW

**Review by shriya-deshmukh — APPROVED** [review]

File: cortx-monitorQuickstartGuide.md [hunk 1]
```diff
@@ -137,23 +137,80 @@ This guide provides a step-by-step walkthrough for getting you CORTX-Monitor Ser
...
    You'll need to create a repository and copy all the required RPMs to it. Run the commands:
 
     ```shell
-    
     $ mkdir MYRPMS
     $ cp -R /root/rpmbuild/RPMS/x86_64/cortx-libsspl_sec-* /root/rpmbuild/RPMS/noarch/cortx-sspl-* MYRPMS
-    $ find MYRPMS -name \*.rpm -print0 | sudo xargs -0 yum install -y
+    $ CORTX_MONITOR_BASE_URL="https://raw.githubusercontent.com/Seagate/cortx-monitor/main"
+    $ curl $CORTX_MONITOR_BASE_URL/low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy -o sspl_dev_deploy
+    $ chmod a+x sspl_dev_deploy
```
Comment (cdeshmukh) [inline cortx-monitorQuickstartGuide.md:150]: keep a clear n bold note that this script is for dev purpose, to independently install SSPL only, along with its dependencies and conflicts with CORTX Provisioning steps meant for whole CORTX stack deployment.

File: cortx-monitorQuickstartGuide.md [hunk 2]
```diff
@@ -137,23 +137,80 @@ This guide provides a step-by-step walkthrough for getting you CORTX-Monitor Ser
...
+    $ ./sspl_dev_deploy --cleanup
+    $ ./sspl_dev_deploy --prereq -L /root/MYRPMS
+    ```
+   
+   Create template file with the name `/root/template_values.1-node.txt`
```
Comment (cdeshmukh) [inline cortx-monitorQuickstartGuide.md:155]: avoid giving /root as the default option. use from ~/

File: cortx-monitorQuickstartGuide.md [hunk 3]
```diff
@@ -137,23 +137,80 @@ This guide provides a step-by-step walkthrough for getting you CORTX-Monitor Ser
...
+    TMPL_CONTROLLER_SECRET=""
+    TMPL_CONTROLLER_TYPE=Gallium
+    ```
+    Make sure to update TMPL_MACHINE_ID, TMPL_HOSTNAME, TMPL_ENCLOSURE_ID, TMPL_BMC_SECRET, TMPL_CONTROLLER_SECRET with actual values.
+    
+    To generate passwords you can use below steps.
+    ```
+    machine_id="xyz"
+    enclosure_id="enc_xyz"
+    bmc_pass='Secret'
```
Comment (cdeshmukh) [inline cortx-monitorQuickstartGuide.md:193]: say 'samplexxxx'

or this 'Secret' will be again picked & flagged by scan

File: cortx-monitorQuickstartGuide.md [hunk 4]
```diff
@@ -137,23 +137,80 @@ This guide provides a step-by-step walkthrough for getting you CORTX-Monitor Ser
...
+    encl_pass='Secret'
+
+    bmc_key=$(python3 -c 'from cortx.utils.security.cipher import Cipher; print(Cipher.generate_key('"'$machine_id'"', "server_node"))')
+    encl_key=$(python3 -c 'from cortx.utils.security.cipher import Cipher; print(Cipher.generate_key('"'$enclosure_id'"', "storage_enclosure"))')
+
+    #Encryption
```
Comment (cdeshmukh) [inline cortx-monitorQuickstartGuide.md:199]: this shows CORTX encryption mechanism, which anyways provisioner would be covering in deployment guide, to enable community users to feed proper encrypted passwords to required config.
Still @johnbent  please confirm if this is fine.

File: cortx-monitorQuickstartGuide.md [hunk 5]
```diff
@@ -28,7 +28,7 @@ This guide provides a step-by-step walkthrough for getting you CORTX-Monitor Ser
     
      1. Run the command:
             
-            `$ curl https://raw.githubusercontent.com/Seagate/cortx-prvsnr/dev/cli/src/cortx-prereqs.sh?token=APA75GY34Y2F5DJSOKDCZAK7ITSZC -o cortx-prereqs.sh; chmod a+x cortx-prereqs.sh`
+            `$ curl https://raw.githubusercontent.com/Seagate/cortx-re/main/scripts/third-party-rpm/install-cortx-prereq.sh?token=APBA7EBWCTPTEOH5PPWFYYTBCN23Y --o cortx-prereqs.sh; chmod a+x cortx-prereqs.sh`
```
Comment (mariyappanp) [inline cortx-monitorQuickstartGuide.md:31]: `curl http://cortx-storage.colo.seagate.com/releases/cortx/third-party-deps/rpm/install-cortx-prereq.sh --o cortx-prereqs.sh; chmod a+x cortx-prereqs.sh`
Link could be more readable.

File: cortx-monitorQuickstartGuide.md [hunk 6]
```diff
@@ -49,8 +49,8 @@ This guide provides a step-by-step walkthrough for getting you CORTX-Monitor Ser
 
         :page_with_curl: **Notes:** 
         
-      - If the https://raw.githubusercontent.com/Seagate/cortx-prvsnr/dev/cli/src/cortx-prereqs.sh?token=APA75GY34Y2F5DJSOKDCZAK7ITSZC link is not accessible, generate new one.
-      - Visit https://github.com/Seagate/cortx-prvsnr/blob/dev/cli/src/cortx-prereqs.sh and naviagte to RAW > Copy URL > Use the URL for deployment.
+      - If the https://raw.githubusercontent.com/Seagate/cortx-re/main/scripts/third-party-rpm/install-cortx-prereq.sh?token=APBA7EBWCTPTEOH5PPWFYYTBCN23Y link is not accessible, generate new one.
+      - Visit https://github.com/Seagate/cortx-re/blob/main/scripts/third-party-rpm/install-cortx-prereq.sh and naviagte to RAW > Copy URL > Use the URL for deployment.
```
Comment (mariyappanp) [inline cortx-monitorQuickstartGuide.md:53]: Why need token? Its dynamic and so token key in the link can be changed later. 
Above I have mentioned generic link. Please check that.

File: cortx-monitorQuickstartGuide.md [hunk 7]
```diff
@@ -137,23 +137,83 @@ This guide provides a step-by-step walkthrough for getting you CORTX-Monitor Ser
...
-
-3. Install CORTX-Monitor RPMs
+    
+5. Install CORTX-Monitor RPMs
 
    You'll need to create a repository and copy all the required RPMs to it. Run the commands:
 
     ```shell
-    
     $ mkdir MYRPMS
```
Comment (mariyappanp) [inline cortx-monitorQuickstartGuide.md:146]: Let this be a full directory path.
`mkdir -p /root/MYRPMS`

File: cortx-monitorQuickstartGuide.md [hunk 8]
```diff
@@ -137,23 +137,83 @@ This guide provides a step-by-step walkthrough for getting you CORTX-Monitor Ser
...
     $ cp -R /root/rpmbuild/RPMS/x86_64/cortx-libsspl_sec-* /root/rpmbuild/RPMS/noarch/cortx-sspl-* MYRPMS
```
Comment (mariyappanp) [inline cortx-monitorQuickstartGuide.md:147]: Keep full path `/root/MYRPMS`

File: cortx-monitorQuickstartGuide.md [hunk 9]
```diff
@@ -137,23 +137,83 @@ This guide provides a step-by-step walkthrough for getting you CORTX-Monitor Ser
...
-    $ find MYRPMS -name \*.rpm -print0 | sudo xargs -0 yum install -y
+    $ CORTX_MONITOR_BASE_URL="https://raw.githubusercontent.com/Seagate/cortx-monitor/main"
+    $ curl $CORTX_MONITOR_BASE_URL/low-level/files/opt/seagate/sspl/setup/sspl_dev_deploy -o sspl_dev_deploy
+    $ chmod a+x sspl_dev_deploy
+    ```
+    **sspl_dev_deploy is only meant for dev purpose and to install SSPL independently along with its dependencies.**
+    ```
+    $ ./sspl_dev_deploy --cleanup
```
Comment (mariyappanp) [inline cortx-monitorQuickstartGuide.md:154]: `--cleanup` includes removing cortx-py-utils RPM.
**Caution**: Any other component RPMS installed on the setup also will get removed while removing cortx-py-utils.

File: cortx-monitorQuickstartGuide.md (hunk 6 above)
Comment (cdeshmukh) [reply cortx-monitorQuickstartGuide.md:53]: agree, lets avoid token when not needed

File: cortx-monitorQuickstartGuide.md (hunk 1 above)
Comment (cdeshmukh) [reply cortx-monitorQuickstartGuide.md:150]: >conflicts with CORTX Provisioning steps meant for whole CORTX stack deployment.

this part is not stated

File: cortx-monitorQuickstartGuide.md (hunk 5 above)
Comment (thavanathan) [reply cortx-monitorQuickstartGuide.md:31]: without token value, curl is giving 404 error.

File: cortx-monitorQuickstartGuide.md (hunk 6 above)
Comment (thavanathan) [reply cortx-monitorQuickstartGuide.md:53]: without token, we are getting 404 error. We have given a note to change is the token if it is not valid anymore.

File: cortx-monitorQuickstartGuide.md (hunk 9 above)
Comment (thavanathan) [reply cortx-monitorQuickstartGuide.md:154]: added a note below the command.

File: cortx-monitorQuickstartGuide.md [hunk 10]
```diff
@@ -49,8 +49,8 @@ This guide provides a step-by-step walkthrough for getting you CORTX-Monitor Ser
 
         :page_with_curl: **Notes:**
```
[the remark this answers is not in the document]
Comment (thavanathan) [reply cortx-monitorQuickstartGuide.md]: Fixed.

File: cortx-monitorQuickstartGuide.md (hunk 6 above)
Comment (jjxsg) [inline cortx-monitorQuickstartGuide.md]: Please correct following typos:
Line 53: naviagte -> navigate
Line 78: 1. Before you build the RPM, you'll need to install the `dependencies` using:

Comment (thavanathan) [reply cortx-monitorQuickstartGuide.md]: Done.

File: cortx-monitorQuickstartGuide.md (hunk 7 above)
Comment (thavanathan) [reply cortx-monitorQuickstartGuide.md]: Done

File: cortx-monitorQuickstartGuide.md (hunk 5 above)
Comment (mariyappanp) [reply cortx-monitorQuickstartGuide.md:31]: Okay.

**Review by jjxsg — APPROVED** [review]

File: cortx-monitorQuickstartGuide.md (hunk 4 above)
Comment (johnbent) [reply cortx-monitorQuickstartGuide.md:202]: Is the secret key merely created by hashing publicly known data?  Shouldn't we include something random or a pass phrase from the user?

Comment (cdeshmukh) [reply cortx-monitorQuickstartGuide.md:202]: Above code is just merely showing a way with which key generation can happen, where for simplicity we are feeding some known data from configuration. As part of Solution encryption logic, consumer is free to feed any input to generate keys.
Probably we should add that as a Note here !? Would be good to relate faster i guess.

@thavanathan can you please add that.

Comment (thavanathan) [reply cortx-monitorQuickstartGuide.md:204]: @cdeshmukh added a note.

**Pull request closed without merging**